internal/command/clusters/uninstall.go

```diff
@@ -0,0 +1,349 @@
+	}
+	if found {
+		certManagerIntegrationsNotification.resourceInfos = append(certManagerIntegrationsNotification.resourceInfos, fmt.Sprintf(integrationInfoTemplate, "cert-manager-csi-driver-spiffe"))
+	}
+	c := &components.CertManagerCSIDriverStatus{}
+	found, err = c.Match(md)
+	if err != nil {
+		return nil, fmt.Errorf("failed to detemine if cert-manager-csi-driver exists: %w", err)
+	}
+	if found {
+		certManagerIntegrationsNotification.resourceInfos = append(certManagerIntegrationsNotification.resourceInfos, fmt.Sprintf(integrationInfoTemplate, "cert-manager-csi-driver"))
+	}
+	i := &components.CertManagerIstioCSRStatus{}
+	found, err = i.Match(md)
+	if err != nil {
+		return nil, fmt.Errorf("failed to detemine if istio-csr exists: %w", err)
+	}
+	if found {
+		certManagerIntegrationsNotification.resourceInfos = append(certManagerIntegrationsNotification.resourceInfos, fmt.Sprintf(integrationInfoTemplate, "cert-manager-istio-csr"))
+	}
+	if len(certManagerIntegrationsNotification.resourceInfos) > 0 {
+		certManagerIntegrationsNotification.header = integrationHeader
+		notifications = append(notifications, certManagerIntegrationsNotification)
+	}
+
+	return notifications, nil
+}
+
+// Notification holds information about a particular type of issue related to
+// uninstallation safety affecting a subset of resources in cluster
+type notification struct {
+	// header holds generic info about the issue and the suggested fix
+	header string
+	// listing of the affected resources with any additional related info
+	resourceInfos []string
+}
+
+func buildClients(kubeconfig *rest.Config) (allClients, error) {
+	secretsClient, err := clients.NewGenericClient[*corev1.Secret, *corev1.SecretList](
+		&clients.GenericClientOptions{
+			RestConfig: kubeconfig,
+			APIPath:    "/api/",
+			Group:      corev1.GroupName,
+			Version:    corev1.SchemeGroupVersion.Version,
+			Kind:       "secrets",
+		},
+	)
+	if err != nil {
+		return allClients{}, fmt.Errorf("error creating new secrets client: %w", err)
+	}
+
+	certsClient, err := clients.NewCertificateClient(kubeconfig)
+	if err != nil {
+		return allClients{}, fmt.Errorf("error creating new cert-manager.io client: %w", err)
+	}
+	podClient, err := clients.NewGenericClient[*corev1.Pod, *corev1.PodList](
+		&clients.GenericClientOptions{
+			RestConfig: kubeconfig,
+			APIPath:    "/api/",
+			Group:      corev1.GroupName,
+			Version:    corev1.SchemeGroupVersion.Version,
+			Kind:       "pods",
+		},
+	)
+	if err != nil {
+		return allClients{}, fmt.Errorf("error creating new pods client: %w", err)
+	}
+	return allClients{
+		certificates: certsClient,
+		secrets:      secretsClient,
+		pods:         podClient,
+	}, nil
+}
+
+type allClients struct {
+	secrets      clients.Generic[*corev1.Secret, *corev1.SecretList]
+	certificates clients.Generic[*cmapi.Certificate, *cmapi.CertificateList]
+	pods         clients.Generic[*corev1.Pod, *corev1.PodList]
+}
+
+func isUnReady(cert cmapi.Certificate) bool {
```

Comment:
Is the name `UnReady` correct?

Reply:
:) no

not sure how I came up with that